deprecate tidb config enable-replica-selector-v2 (#18026)
pingcap/docs: tidb-configuration-file.md

> **Warning:**
>

> > Since v8.2.0, this configuration item is deprecated. New version of the Region replica selector will be used by default when sending RPC requests to TiKV.

Reviewer: ```suggestion
> Starting from v8.2.0, this configuration item is deprecated. The new version of the Region replica selector is used by default when sending RPC requests to TiKV.
```
